nifi-nar-bundles/nifi-extension-utils/nifi-file-transfer/src/main/java/org/apache/nifi/processor/util/file/transfer/FileTransfer.java

```diff
@@ -172,6 +173,8 @@ default String getAbsolutePath(FlowFile flowFile, String remotePath) throws IOEx
             + "than normal.")
         .defaultValue("5000")
         .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+        // Limiting to batches means that TIME_TRACKING and TIME_WINDOW strategies will leave files behind
+        .dependsOn(AbstractListProcessor.LISTING_STRATEGY, AbstractListProcessor.BY_ENTITIES, AbstractListProcessor.NO_TRACKING)
```

Comment:
This should not be changed because it also impacts the `ListFTP` Processor.

Reply:
I did intend for the changes to cover both, I can't think how `TIME_TRACKING` can work with batches for either `ListSFTP` or `ListFTP`. Have you got an example where it does?

I checked out `GetFTP` and `GetSFTP` and they retain the property even with the dependency, it seems that if the 'depended on' property doesn't exist then it defaults to visible.

Happy to make the change of course, can you think of a different way to communicate the limitation to the user?